File: arrays/shuffleTheArray.py

class Solution(object):
    def shuffle(self, nums, n):

        #List Slicing
        #Max with this
        # Runtime: 52 ms, faster than 96.86% of Python3 online submissions for Shuffle the Array.
        # Memory Usage: 14.2 MB, less than 92.90% of Python3 online submissions for Shuffle the Array.
        
        neww=[]
        new1=nums[:n]
        new2=nums[n:]
        for i in range(n):
            neww.append(new1[i])
            neww.append(new2[i])
        return neww
        
        

        #Runtime: 52 ms, faster than 96.86% of Python3 online submissions for Shuffle the Array.
        #Memory Usage: 14.3 MB, less than 78.23% of Python3 online submissions for Shuffle the Array.

#         arr=[]
#         for i in range(n):
#            arr.append(nums[i])
#            arr.append(nums[i+n])
#         return arr
        
    
        #two pointer
        
        # p1,p2,neww=0,n,[]
        # while(p1<n):
        #     neww.append(nums[p1])
        #     neww.append(nums[p2])
        #     p1+=1
        #     p2+=1
        # return neww
    
    
    
        # One liner solution
        # return [j for i in range(n) for j in (nums[i], nums[i+n])]
    
        ''' One liner Explanation'''
        
        '''
        
        for i in range(n):
            for j in (nums[i], nums[i+n]):
                print(j)
        '''
    
    

    
print(Solution.shuffle(Solution, [2,5,1,3,4,7], 3))
    

File: arrays/shuffleTheArray.java
package arrays;

import java.util.Arrays;

public class shuffleTheArray {
    public static void main(String[] args) {
        SolutionSTA solution = new SolutionSTA();
        System.out.println(Arrays.toString(solution.shuffle(new int[]{1,2,3,4,4,3,2,1}, 4))); //[2,3,5,4,1,7] 
        System.out.println(Arrays.toString(solution.shuffle(new int[]{2,5,1,3,4,7}, 3))); //[1,4,2,3,3,2,4,1]
        System.out.println(Arrays.toString(solution.shuffle(new int[]{1,1,2,2}, 2))); //[1,2,1,2]

    }    
}


class SolutionSTA {
    public int[] shuffle(int[] nums, int n) {
        int [] result = new int [nums.length];
        for(int i=0 ; i< n; i++){
            result[i*2]= nums[i];
            result[i*2+1]=nums[i+n];
        }
        return result;
    }
}